MHArmory.Search/OpenCL/Host.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using System.Collections.ObjectModel;
+using System.Text;
+using Cloo;
+
+namespace MHArmory.Search.OpenCL
+{
+    class Host : IDisposable
+    {
+        private ComputeContext Context { get; }
+        private ComputeProgram Program { get; }
+        private const string KernelName = "search";
+
+        public Host()
+        {
+            byte[] source = TrimUTFBOM(Properties.Resources.search);
```

Comment:
The method `TrimUTFBOM` makes an allocation and copy. Instead it is simpler to ensure file `search.cl` does not include BOM and remove that method. If you absolutely want to keep it, you should make it return an `ArraySegmnet<byte>` in order to avoid allocation and copy, then the `Encoding.ASCII.GetString()` call can make use of array and indices.

Reply:
The text editors add it automatically, or maybe it comes from the resource embedding, I'm not even sure. Point is, even if I remove it somehow, someone else might come along, use another text editor, and it comes right back, and now the kernel doesn't compile. I think it's much better to just remove it programmatically.

I'll return an array segment like you suggested.